sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableDropPartitionSuite.scala

```diff
@@ -30,17 +30,24 @@ class AlterTableDropPartitionSuite
   with CommandSuiteBase {
 
   test("hive client calls") {
-    withNamespaceAndTable("ns", "tbl") { t =>
-      sql(s"CREATE TABLE $t (id int, part int) $defaultUsing PARTITIONED BY (part)")
-      sql(s"INSERT INTO $t PARTITION (part=0) SELECT 0")
-      sql(s"INSERT INTO $t PARTITION (part=1) SELECT 1")
-
-      checkHiveClientCalls(expected = 19) {
-        sql(s"ALTER TABLE $t DROP PARTITION (part=0)")
-      }
-      sql(s"CACHE TABLE $t")
-      checkHiveClientCalls(expected = 22) {
-        sql(s"ALTER TABLE $t DROP PARTITION (part=1)")
+    Seq(false, true).foreach { statsOn =>
+      withSQLConf(SQLConf.AUTO_SIZE_UPDATE_ENABLED.key -> statsOn.toString) {
+        withNamespaceAndTable("ns", "tbl") { t =>
+          sql(s"CREATE TABLE $t (id int, part int) $defaultUsing PARTITIONED BY (part)")
+          sql(s"INSERT INTO $t PARTITION (part=0) SELECT 0")
+          sql(s"INSERT INTO $t PARTITION (part=1) SELECT 1")
+          sql(s"ALTER TABLE $t ADD PARTITION (part=2)") // empty partition
+          checkHiveClientCalls(expected = if (statsOn) 27 else 19) {
+            sql(s"ALTER TABLE $t DROP PARTITION (part=2)")
+          }
+          checkHiveClientCalls(expected = if (statsOn) 32 else 19) {
```

Comment:
I could understand that we don't care of DDL command performance so much but 32 calls to the catalog (the calls can go over network to external dbms). @cloud-fan @HyukjinKwon @dongjoon-hyun Isn't it too much?

Reply:
yea we should improve it.